File: dart-sdk/test/integration/vobiz_integration_test.dart
import 'dart:io';
import 'package:vobiz/api.dart';

/// Vobiz Dart SDK - Integration Tests (Read-Only)
/// Run: dart test/integration/vobiz_integration_test.dart
/// Requires: VOBIZ_AUTH_ID and VOBIZ_AUTH_TOKEN env vars
Future<void> main() async {
  final authId    = Platform.environment['VOBIZ_AUTH_ID'] ?? '';
  final authToken = Platform.environment['VOBIZ_AUTH_TOKEN'] ?? '';

  if (authId.isEmpty || authToken.isEmpty) {
    print('SKIP: VOBIZ_AUTH_ID or VOBIZ_AUTH_TOKEN not set');
    exit(0);
  }

  final client = ApiClient(basePath: 'https://api.vobiz.ai');
  client.addDefaultHeader('X-Auth-ID', authId);
  client.addDefaultHeader('X-Auth-Token', authToken);

  int passed = 0;
  int failed = 0;

  Future<void> runTest(String name, Future<void> Function() fn) async {
    try {
      await fn();
      print('[Dart] PASS: $name');
      passed++;
    } catch (e) {
      print('[Dart] FAIL: $name - $e');
      failed++;
    }
  }

  final accountApi    = AccountApi(client);
  final callApi       = CallApi(client);
  final recordingApi  = RecordingApi(client);
  final conferenceApi = ConferenceApi(client);
  final appApi        = ApplicationApi(client);

  await runTest('GetAccountDetails', () async {
    final result = await accountApi.apiV1AuthMeGet();
    print('  -> Response received OK');
  });

  await runTest('GetLiveCalls', () async {
    final result = await callApi.apiV1AccountAuthIdCallGet(authId, status: 'live');
    print('  -> Response received OK');
  });

  await runTest('ListRecordings', () async {
    final result = await recordingApi.apiV1AccountAccountIdRecordingGet(authId);
    print('  -> Response received OK');
  });

  await runTest('ListConferences', () async {
    final result = await conferenceApi.apiV1AccountAuthIdConferenceGet(authId);
    print('  -> Response received OK');
  });

  await runTest('ListApplications', () async {
    final result = await appApi.apiV1AccountAuthIdApplicationGet(authId);
    print('  -> Response received OK');
  });

  print('\n[Dart] Results: $passed passed, $failed failed');
  exit(failed > 0 ? 1 : 0);
}

File: kotlin-sdk/src/test/kotlin/ai/vobiz/integration/VobizIntegrationTest.kt
package ai.vobiz.integration

import ai.vobiz.apis.*
import ai.vobiz.infrastructure.ApiClient

/**
 * Vobiz Kotlin SDK - Integration Tests (Read-Only)
 * Run: ./gradlew integrationTest
 * Requires: VOBIZ_AUTH_ID and VOBIZ_AUTH_TOKEN env vars
 */
fun main() {
    val authId    = System.getenv("VOBIZ_AUTH_ID")
    val authToken = System.getenv("VOBIZ_AUTH_TOKEN")

    if (authId.isNullOrEmpty() || authToken.isNullOrEmpty()) {
        println("SKIP: VOBIZ_AUTH_ID or VOBIZ_AUTH_TOKEN not set")
        return
    }

    val client = ApiClient(
        baseUrl = "https://api.vobiz.ai",
        bearerToken = null
    ).apply {
        addDefaultHeader("X-Auth-ID", authId)
        addDefaultHeader("X-Auth-Token", authToken)
    }

    var passed = 0
    var failed = 0

    fun runTest(name: String, block: () -> Unit) {
        try {
            block()
            println("[Kotlin] PASS: $name")
            passed++
        } catch (e: Exception) {
            println("[Kotlin] FAIL: $name - ${e.message}")
            failed++
        }
    }

    runTest("GetAccountDetails") {
        val api = AccountApi(client)
        val result = api.apiV1AuthMeGet()
        println("  -> Response received OK")
    }

    runTest("GetLiveCalls") {
        val api = CallApi(client)
        val result = api.apiV1AccountAuthIdCallGet(authId, status = "live")
        println("  -> Response received OK")
    }

    runTest("ListRecordings") {
        val api = RecordingApi(client)
        val result = api.apiV1AccountAccountIdRecordingGet(authId)
        println("  -> Response received OK")
    }

    runTest("ListConferences") {
        val api = ConferenceApi(client)
        val result = api.apiV1AccountAuthIdConferenceGet(authId)
        println("  -> Response received OK")
    }

    runTest("ListApplications") {
        val api = ApplicationApi(client)
        val result = api.apiV1AccountAuthIdApplicationGet(authId)
        println("  -> Response received OK")
    }

    println("\n[Kotlin] Results: $passed passed, $failed failed")
    if (failed > 0) System.exit(1)
}
